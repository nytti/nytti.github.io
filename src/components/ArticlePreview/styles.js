export default {
  action: {
    width: '100%',
  },
  media: {
    height: 220,
  },
  link: {
    textDecoration: 'none',
  }
}
