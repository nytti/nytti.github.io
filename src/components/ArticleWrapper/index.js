// News Item wrapper used for generalizing state selector
// It's used for both preview and separate page articles
// So it's receiving component for rendering

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Selectos
import { getArticle } from 'selectors/article'

// Components
import ArticleNotFound from 'components/ArticleNotFound'

const mapStateToProps = (state, { id }) => ({
  article: getArticle(state, { id })
})

export default
@connect(mapStateToProps)
class ArticleWrapper extends Component {
  static propTypes = {
    article: PropTypes.object,
    topic: PropTypes.string,
    component: PropTypes.func.isRequired,
  }

  render() {
    const { topic, article, component: Render } = this.props
    if (!article) {
      return <ArticleNotFound />
    }
    return (
      <Render
        key={article._id}
        article={article}
        topic={topic}
      />
    )
  }
}
