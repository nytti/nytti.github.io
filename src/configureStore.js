import { createStore, combineReducers, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { routerReducer, routerMiddleware } from 'react-router-redux'
import { gettersMiddleware } from 'utils/reduxGetters'

import * as reducers from './reducers'

export default (history, ...middlewares) => {
  const createReducer = () => combineReducers({
    router: routerReducer,
    ...reducers,
  })

  const store = createStore(
    createReducer(),
    applyMiddleware(
      thunk,
      routerMiddleware(history),
      gettersMiddleware,
      ...middlewares,
    )
  )

  if (module.hot) {
    module.hot.accept('./reducers', () => store.replaceReducer(createReducer()))
  }

  return store
}
