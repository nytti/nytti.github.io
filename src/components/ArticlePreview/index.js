// One item in news list. This component is stateful and little tricky -
// it handles news item modal dialog rendering to make system architecture
// more encapsulated

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

// Material UI
import { withStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardMedia from '@material-ui/core/CardMedia'
import CardContent from '@material-ui/core/CardContent'
import CardActionArea from '@material-ui/core/CardActionArea'
import Typography from '@material-ui/core/Typography'

import { nyt } from 'config/config'

// Styles
import styles from './styles'

const { imageUrl } = nyt

export default
@withStyles(styles)
class ArticlePreview extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    article: PropTypes.object.isRequired,
    topic: PropTypes.string.isRequired,
  }

  render() {
    const { article, classes, topic } = this.props
    const {
      _id: id,
      multimedia,
      headline: {
        main: title,
      },
    } = article
    const media = multimedia.find(({ subType }) => subType === 'master768') || multimedia[0]

    return (
      <Card className={classes.root}>
        <Link to={`/${id}`} className={classes.link}>
          <CardActionArea className={classes.action}>
            <CardContent>
              <Typography>
                { title }
              </Typography>
            </CardContent>
            { media && (
              <CardMedia
                className={classes.media}
                image={`${imageUrl}${media.url}`}
                title={title}
              />
            )}
            <CardContent>
              <Typography component="p" color="textSecondary">
                { topic }
              </Typography>
            </CardContent>
          </CardActionArea>
        </Link>
      </Card>
    )
  }
}
