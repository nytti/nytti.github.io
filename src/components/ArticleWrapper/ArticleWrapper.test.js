import React from 'react'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router'
import thunk from 'redux-thunk'

import state from 'articles/state'

import items from 'articles/items'

import ArticlePreview from 'components/ArticlePreview'
import ArticleWrapper from './index'

// Mocking store with test data
const mockStore = configureStore([thunk])
const store = mockStore(state)

describe('<ArticleWrapper />', () => {
  test('<ArticleWrapper /> snapshot', () => {
    const element = shallow(
      <ArticleWrapper store={store} id={items[0]._id} component={ArticlePreview} />
    )
    expect(element).toMatchSnapshot()
  })

  test('<ArticleWrapper /> mount with store', () => {
    const component = mount(
      <MemoryRouter initialEntries={[`/${items[0]._id}`]}>
        <ArticleWrapper topic="-topic" store={store} id={items[0]._id} component={ArticlePreview} />
      </MemoryRouter>
    )
    expect(component.text()).toBe(`${items[0].headline.main}-topic`)
  })
})
