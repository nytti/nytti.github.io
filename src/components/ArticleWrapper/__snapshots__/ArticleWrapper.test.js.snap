// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleWrapper /> <ArticleWrapper /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ArticleWrapper)
    component={[Function]}
    id="Item-Id-Test"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "article": Object {
        "_id": "Item-Id-Test",
        "document_type": "article",
        "headline": Object {
          "main": "Item-Headline-Main-Test",
        },
        "multimedia": Array [
          Object {
            "url": "image-url-test",
          },
        ],
        "pub_date": "2018-09-24T15:58:39+0000",
        "snippet": "Item-Snippet-Test",
        "web_url": "https://test-item.com",
      },
      "component": [Function],
      "dispatch": [Function],
      "id": "Item-Id-Test",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "article": Object {
          "_id": "Item-Id-Test",
          "document_type": "article",
          "headline": Object {
            "main": "Item-Headline-Main-Test",
          },
          "multimedia": Array [
            Object {
              "url": "image-url-test",
            },
          ],
          "pub_date": "2018-09-24T15:58:39+0000",
          "snippet": "Item-Snippet-Test",
          "web_url": "https://test-item.com",
        },
        "component": [Function],
        "dispatch": [Function],
        "id": "Item-Id-Test",
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
